Integrations/python/deephaven/learn/__init__.py

```diff
@@ -0,0 +1,190 @@
+        _Scatterer_ = jpy.get_type("io.deephaven.integrations.learn.Scatterer")
+
+
+# every module method that invokes Java classes should be decorated with @_passThrough
+@wrapt.decorator
+def _passThrough(wrapped, instance, args, kwargs):
+    """
+    For decoration of module methods, to define necessary symbols at runtime
+
+    Args:
+        wrapped  : the method to be decorated
+        instance : the object to which the wrapped function was bound when it was called
+        args     : the argument list for `wrapped`
+        kwargs   : the keyword argument dictionary for `wrapped`
+
+    Returns:
+        the decorated version of the method.
+    """
+    _defineSymbols()
+    return wrapped(*args, **kwargs)
+
+
+@_passThrough
+class Input:
+    """
+    Input specifies how to gather data from a Deephaven table into a Python object.
+    """
+    def __init__(self, colNames, gatherFunc):
+        """
+        Args:
+            colNames   : column name or list of column names from which to gather input.
+            gatherFunc : python function that determines how input gets transformed into a python object.
+        """
+        self.input = _Input_(colNames, gatherFunc)
+
+    def __str__(self):
+        return "Columns: " + str([self.input.getColNames()[i] for i in range(len(self.input.getColNames()))])
+
+
+@_passThrough
+class Output:
+    """
+    Output specifies how to scatter data from a Python object into a table column.
+    """
+    def __init__(self, colName, scatterFunc, type=None):
+        """
+        Args:
+            colName         : name of the new column that will store results.
+            scatterFunc     : python function that determines how data is taken from a python object into a Deephaven table.
+            type (optional) : desired data type of the new output column. Must be one of int, long, double, float, boolean, char, String.
+        """
+        self.output = _Output_(colName, scatterFunc, type)
+
+    def __str__(self):
+        print("Column: " + self.output.getColName())
+        print("Type: " + self.output.getType())
+
+
+@_passThrough
+def _validate(inputs, outputs, table):
+    """
+    Ensures that all input columns exist in the table, and that no output column names already exist in the table.
+
+    Args:
+        inputs  : list of Inputs to validate.
+        outputs : list of Outputs to validate.
+        table   : table to check Input and Output columns against.
+
+    Raises:
+        ValueError : if at least one of the Input columns does not exist in the table.
+        ValueError : if at least one of the Output columns already exists in the table.
+    """
+
+    inputColumns = [input.input.getColNames()[i] for input in inputs for i in range(len(input.input.getColNames()))]
+
+    if not all(inputColumn in list(table.getMeta().getColumn("Name").getDirect()) for inputColumn in inputColumns):
+        raise ValueError("All Input columns must exist in the given table.")
```

Comment:
Error message is not helpful.  You need to create a list of problematic columns.  I suspect this is better accomplished via set arithmetic to do the check.  Then if the residual set is not empty, error and print the residual set.  Make it very explicit to the user what is wrong.